clang/include/clang/Sema/Sema.h

```diff
@@ -13125,6 +13125,19 @@ class Sema final {
                                    SourceLocation BuiltinLoc,
                                    SourceLocation RParenLoc);
 
+  template <typename AttrTy>
+  bool isSyclGlobalType(QualType Ty) {
```

Comment:
How about something like
```suggestion
  bool isDecoratedWithSyclAttribute(QualType Ty) {
```
So if there is more attributes we can use this function for them as well.